packages/hw-app-eth/src/Eth.ts

```diff
@@ -197,14 +197,40 @@ export default class Eth {
     const toSend: Buffer[] = [];
     let response;
     // Check if the TX is encoded following EIP 155
-    let rlpTx = ethers.utils.RLP.decode(rlpData).map((hex) =>
+    const rlpTx = ethers.utils.RLP.decode(rlpData).map((hex) =>
       Buffer.from(hex.slice(2), "hex")
     );
 
     let vrsOffset = 0;
     let chainId = new BigNumber(0);
     let chainIdTruncated = 0;
 
+    const rlpDecoded = ethers.utils.RLP.decode(rlpData);
+
+    let decodedTx;
+    if (txType === 2) {
+      // EIP1559
+      decodedTx = {
+        data: rlpDecoded[7],
+        to: rlpDecoded[5],
+        chainId: rlpTx[0],
+      };
+    } else if (txType === 1) {
+      // EIP2930
+      decodedTx = {
+        data: rlpDecoded[6],
+        to: rlpDecoded[4],
+        chainId: rlpTx[0],
+      };
+    } else {
+      // Legacy tx
+      decodedTx = {
+        data: rlpDecoded[5],
+        to: rlpDecoded[3],
+        chainId: rlpTx[6],
```

Comment:
`rlpTx[6]` is only valid for post EIP-155 transactions (see https://eips.ethereum.org/EIPS/eip-155)
If `rlpTx.length < 6` then the `chainId` should default to `1` :)

Reply:
Will change 👍